State_manager_code/StateManager.py

```diff
@@ -37,12 +37,11 @@ def __init__(self):
             States.Start : [States.CenterReward],
             States.CenterReward : [States.TrialStarted],
             States.TrialStarted : [States.M1CM2C, States.M1CM2D, States.M1DM2C, States.M1DM2D],
-            States.M1CM2C : [States.WaitForReturn],
-            States.M1CM2D: [States.WaitForReturn],
-            States.M1DM2C: [States.WaitForReturn],
-            States.M1DM2D: [States.WaitForReturn],
-            States.WaitForReturn:[States.TrialCompleted],
-            States.TrialCompleted:[States.End],
+            States.M1CM2C : [States.TrialCompleted],
```

Comment:
Please remove the WaitForReturn state from the States enum (and change the numbers accordingly)
Also please change the name TrialAbort to ReturnAbort (The Trial have just been completed so it cannot be aborted).
Talking about names, I think we should change ReturnAbort to ReturnTimeOut and DecisionAbort to DecisionTimeOut. What say you?

Reply:
yes the nomencaluture you suggest more fitting and i will remove waitforreturn

Reply:
Good Morning Anushka, 
Sorry for being a bit of nuisance. I gave the states some more thoughts and I think you were right in two things.

1) it will be more readable if we added a new event for ExperimentStopped - rather then use LastTrial event. (same behavior, easier to understand what's going on).
2) In the ReturnAbort and DecisionAbort we do need transitions to End. otherwise if the mouse never returns to center the program will get stuck.

I sent you a new state diagram via email. What do you think?

I used ReturnAbort and DecisionAbort - but if you prefer Timeout, so be it. :-)

Reply:
i prefer timeout now that i have already changes the states :)

Reply:
Timeout it is